### PR TITLE
Update rules on how rgw_count is determined based on OCS version

### PR DESCRIPTION
**Pull request by clacroix12**
Fixes: #3663

Signed-off-by: Coady LaCroix <clacroix@redhat.com>

### CODE REVIEW

**Comment by clacroix12** [comment]
@ebenahar We had a few different places in the code where we were computing the `rgw_count`. I have moved the calculation to a utility module and am now calling that utility so we have one point of truth for what the `rgw_count` should be in any given deployment/upgrade scenario. I have also added a parametrized unit test to ensure the function is returning what we expect. I believe this meets the requirements we talked about but please let me know if I misunderstood any of the rules regarding the rgw_count, especially around the removal of the azure specific calculation. Also feel free to kick off a PR validation or let me know what to run. I wasn't entirely sure which scenarios you would want to verify this PR with.

@nehaberry I wanted to tag you on this as well since I believe you wrote some of the original `rgw_count` code. Please let me know if this doesn't look right.

File: ocs_ci/utility/rgwutils.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def get_rgw_count(ocs_version, is_upgrade, version_before_upgrade):
```
Comment (ebenahar) [inline ocs_ci/utility/rgwutils.py:6]: good idea to have this logic done in a function

File: ocs_ci/ocs/ocs_upgrade.py [hunk 2]
```diff
@@ -123,28 +124,15 @@ def verify_image_versions(old_images, upgrade_version, version_before_upgrade):
     )
     verify_pods_upgraded(old_images, selector=constants.MDS_APP_LABEL, count=2)
     if config.ENV_DATA.get("platform") in constants.ON_PREM_PLATFORMS:
-        # RGW count is 1 if the cluster was upgraded from <= 4.4
-        # Related bug - https://bugzilla.redhat.com/show_bug.cgi?id=1857802
-        rgw_count = 2 if float(version_before_upgrade) >= 4.5 else 1
+        rgw_count = get_rgw_count(
+            float(upgrade_version), True, float(version_before_upgrade)
```
Comment (petr-balogh) [inline ocs_ci/ocs/ocs_upgrade.py:128]: What I've heard from yesterday's meeting that the upgrade from 4.6 to 4.7 that it should scale down RGW pods to 1.  I am testing it now here:  https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-trigger-vsphere-upi-1az-rhcos-vsan-3m-3w-upgrade-ocs-auto-nightly/15/console so we can check on this cluster. I am pausing before teardown as I would like to verify also another issue when we don't se noobaa core pod after upgrade.

File: ocs_ci/utility/rgwutils.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        ocs_version (str): OCS Version
+        is_upgrade (bool): If cluster was upgraded to current version
+        version_before_upgrade (str): OCS Version prior to upgrade
+
+    Returns:
+        int: RGW Count
+
+    """
+
+    # Assume upgrade from prior version if one is not provided
```
[the remark this answers is not in the document]
Comment (clacroix12) [reply ocs_ci/utility/rgwutils.py:27]: Yeah the way I have it now is that the caller needs to explicitly pass `None` for this parameter (see mcg.py). I could default it to `None` if we think that would be better.

File: ocs_ci/utility/rgwutils.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    if is_upgrade:
+        log.debug("version_before_upgrade not provided, assuming prior release.")
+        version_before_upgrade = version_before_upgrade or round(
+            float(ocs_version) - 0.1, 1
```
[the remark this answers is not in the document]
Comment (clacroix12) [reply ocs_ci/utility/rgwutils.py:31]: We were already casting the ocs_version value to a float with the old logic for rgw_count so either we didn't hit that or it was not an issue previously. We can test it out but I don't believe we ever include `-ga` in the `ocs_version` specified in the config. Please correct me if I'm wrong.

File: ocs_ci/ocs/ocs_upgrade.py (hunk 2 above)
Comment (clacroix12) [reply ocs_ci/ocs/ocs_upgrade.py:128]: Was able to confirm with this execution that the rgw count scales down to 1 with an upgrade from 4.6 to 4.7. I'll update the function logic and the unit test to match this.

File: ocs_ci/utility/rgwutils.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (petr-balogh) [reply ocs_ci/utility/rgwutils.py:30]: We don't add any -ga to OCS version, it's always like x.y. https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/framework/conf/ocs_version/ocs-4.6.yaml#L8

**Review by petr-balogh — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]

**Comment by ebenahar** [comment]
Triggered OCS upgrade - https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-trigger-test-pr/3726/

**Comment by petr-balogh** [comment]
Triggered another upgrade job here:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-trigger-test-pr/3727/console

As it failed on one issue introduced in jenkins job

**Comment by petr-balogh** [comment]
@ebenahar upgrade is now failing cause of this BZ:  https://bugzilla.redhat.com/show_bug.cgi?id=1900722 so I suggest to merge this one and in case something else will be wrong with upgrade we will cover in next PR. Anyway upgrade from 4.6 to 4.7 is failing already for long time cause of different BZs we opened.

**Pull request merged**